### PR TITLE
Set quantity value when stock tracking is enabled

### LINKED ISSUE #37117: [Enhancement]: Set quantity value when stock tracking is enabled
When stock tracking is enabled for the first time, we default to quantity=1 (now we default to 0, which creates lots of issues).

screenshot-user-images githubusercontent com-2023 03 07-22_00_02 (2)

**Designs:**
[zs56kcy53K7GoGhgyjcQ2R-fi-210%3A30925](https://href.li/?https://www.figma.com/file/zs56kcy53K7GoGhgyjcQ2R/?node-id=210%3A30925)

```[tasklist]
### Acceptance criteria
- [ ] Set quantity = 1 when stock tracking is enabled for the first time.
- [ ] Rename stock quantity label from `Stock quantity` to `Quantity`.
```

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

When stock tracking is enabled for the first time, we default to quantity=1 (now we default to 0, which creates lots of issues).

screenshot-user-images githubusercontent com-2023 03 07-22_00_02 (2)

Closes #37117.

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

#### Default value

1. Go to `Products` > `Add New` > `Product data` > `Inventory`.
2. Check the `Stock management` checkbox.
3. Verify that the `Stock quantity` label now says `Quantity` and its value is `1`.
4. Publish the product.
5. Verify that the quantity is `1`.

#### Changing to a different value

1. Go to `Products` > `Add New` > `Product data` > `Inventory`.
2. Check the `Stock management` checkbox.
3. Verify that the `Stock quantity` label now says `Quantity` and its value is `1`.
4. Change the quantity to `3`.
5. Publish the product.
6. Verify that the quantity is `3`.
7. Change the quantity to `7`.
8. Save the updated product.
9. Verify that the quantity is `7`.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php [hunk 1]
```diff
@@ -50,8 +50,8 @@
 			woocommerce_wp_text_input(
 				array(
 					'id'                => '_stock',
-					'value'             => wc_stock_amount( $product_object->get_stock_quantity( 'edit' ) ),
-					'label'             => __( 'Stock quantity', 'woocommerce' ),
+					'value'             => wc_stock_amount( $product_object->get_stock_quantity( 'edit' ) ?? 1 ),
```
Comment (octaedro) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php:53]: I added the changes here instead of the constructor to ensure we won't have any unexpected side effects.

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php [hunk 2]
```diff
@@ -62,7 +62,7 @@
 				)
 			);
 
-			echo '<input type="hidden" name="_original_stock" value="' . esc_attr( wc_stock_amount( $product_object->get_stock_quantity( 'edit' ) ) ) . '" />';
+			echo '<input type="hidden" name="_original_stock" value="' . esc_attr( wc_stock_amount( $product_object->get_stock_quantity( 'edit' ) ?? 1 ) ) . '" />';
```
Comment (mattsherman) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php]: This will set the original stock value to `1` even when it really is `0`. And results in an error message when saving, since we are altering the "original" value. I think we can just revert this change and things will work properly (I tried it locally).

Screenshot 2023-03-21 at 12 41 54

**Review by mattsherman — CHANGES_REQUESTED** [review]
Unable to save product correctly.

I believe if we don't change the "original" value (see comment in code) that things will work properly.

**Comment by mattsherman** [comment]
@octaedro Updated the test instructions to include saving the product and editing a saved product.

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php (hunk 2 above)
Comment (octaedro) [reply plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-inventory.php]: Thank you for the review. I'm still confused about how I managed to test this PR before pushing it. I guess I used the wrong branch 🤦🏻‍♂️

The problem was that I was setting the `_original_stock` as `1` but during the check [when saving](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php#L344) it was getting a `0`, and this is why we saw the error.
I fixed that in commit [02ca669](https://github.com/woocommerce/woocommerce/pull/37304/commits/02ca66943bda161cf4c042435aff1b6e4e36fadd). The `_original_stock` shouldn't change (as you mentioned).

**Review by mattsherman — APPROVED** [review]
Much, much better! 😄

Looks good to me now. :shipit:
